### PR TITLE
[macOS] Replace all `Alt/Option+Letter/Number` default shortcuts to avoid conflicts with special character input.

### LINKED ISSUE #66678: Cannot type @ in the editor on macOS
### Godot version

4.0.dev

### System information

Macbook Air 2021

### Issue description

I am unable to type @ to the editor because command + 2 is bound to switching tabs (2d, 3d, script...)

### Steps to reproduce

Try to type @ on a mac

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/issues/66678 and other similar issues.

### CODE REVIEW

**Comment by akien-mga** [comment]
Meta+Ctrl+Alt+Key combinations seem quite convoluted. Can't we find other combinations without Alt if it's too conflicting with normal input?

Otherwise, it seems like not setting any shortcut might be better than shortcuts which can't be discovered nor typed easily.

**Comment by bruvzg** [comment]
> Meta+Ctrl+Alt+Key combinations seem quite convoluted. Can't we find other combinations without Alt if it's too conflicting with normal input?

Simplified it a bit (with one exception):

| Action | Default Shortcut | macOS Shortcut | Note |
| -- | -- | -- | -- |
| ui_text_select_word_under_caret | ALT+G | META+CTRL+G | |
| file_dialog/toggle_favorite | ALT+F | META+CTRL+F | |
| file_dialog/toggle_mode | ALT+V | META+CTRL+V | |
| script_text_editor/toggle_fold_line | ALT+F | CTRL+META+F | |
| editor/quick_open | SHIFT+ALT+O | META+CTRL+O | |
| editor/quit_to_project_list | CTRL+SHIFT+Q | META+CTRL+ALT+Q | everything with a Q is either used or system keys |
| editor/editor_2d | CTRL+F1 | META+CTRL+KEY_1 | |
| editor/editor_3d | CTRL+F2 | META+CTRL+KEY_2 | |
| editor/editor_script | CTRL+F3 | META+CTRL+KEY_3 | |
| editor/editor_assetlib | CTRL+F4 | META+CTRL+KEY_4 | |
| script_editor/save_all | SHIFT+ALT+S | META+CTRL+S | |

**Review by akien-mga — APPROVED** [review]

File: modules/mono/csharp_script.cpp [hunk 1]
```diff
@@ -1147,6 +1147,7 @@ void CSharpLanguage::_editor_init_callback() {
 	// Add plugin to EditorNode and enable it
 	EditorNode::add_editor_plugin(godotsharp_editor);
 	ED_SHORTCUT("mono/build_solution", TTR("Build Solution"), KeyModifierMask::ALT | Key::B);
+	ED_SHORTCUT_OVERRIDE("mono/build_solution", "macos", KeyModifierMask::META | KeyModifierMask::CTRL | KeyModifierMask::ALT | Key::B);
```
Comment (raulsntos) [inline modules/mono/csharp_script.cpp:1150]: You didn't mention this shortcut in the previous table, is it fine to keep it as is or should it be simplified to META+CTRL+B?

Comment (bruvzg) [reply modules/mono/csharp_script.cpp]: Indeed, I have missed it. Updated to `META+CTRL+B`.

**Comment by akien-mga** [comment]
With this PR merged locally (and a few others, but this one seems like the likely culprit), I get these errors when opening the editor:
```
ERROR: EditorSettings not instantiated yet.
   at: ED_SHORTCUT_OVERRIDE (editor/editor_settings.cpp:1451)
ERROR: EditorSettings not instantiated yet.
   at: ED_SHORTCUT_OVERRIDE (editor/editor_settings.cpp:1451)
ERROR: EditorSettings not instantiated yet.
   at: ED_SHORTCUT_OVERRIDE (editor/editor_settings.cpp:1451)
ERROR: EditorSettings not instantiated yet.
   at: ED_SHORTCUT_OVERRIDE (editor/editor_settings.cpp:1451)
```

**Comment by bruvzg** [comment]
> With this PR merged locally (and a few others, but this one seems like the likely culprit), I get these errors when opening the editor

Seems like help generator is creating `EditorFileDialog` before editor settings.

**Comment by bruvzg** [comment]
Hm, seems like `ED_SHORTCUT_OVERRIDE` is completely broken, all overridden shortcuts are reset on editor restart.

**Comment by akien-mga** [comment]
I was a bit dubious of that change, might be related? #71792

**Comment by bruvzg** [comment]
> I was a bit dubious of that change, might be related?

Yes, reverting this change fix it.

**Comment by akien-mga** [comment]
CC @EricEzaM @YuriSizov

**Comment by bruvzg** [comment]
> I was a bit dubious of that change, might be related? https://github.com/godotengine/godot/pull/71792

Updated PR. Should fix both #71792 regression and #66428.

**Comment by akien-mga** [comment]
Looks good to me :+1:

**Comment by YuriSizov** [comment]
Oh, sorry about that. I'm not too familiar with that part, but the changed look sensible at the time.

**Comment by akien-mga** [comment]
Thanks!

**Comment by EricEzaM** [comment]
Hi All - sorry about that. I messed up here.

However, why was this removed from `editor_settings.cpp`?
```cpp
sc->set_meta("original", events.duplicate(true)); //to compare against changes
```

Now if a shortcut is loaded from the editor settings file (e.g. a user has customised a shortcut), then there is nothing to compare against for reverting it. This also breaks the shortcut list as it expects shortcuts to have an "original" meta set.

Go into editor settings, customise a shortcut (that does not have a platform `ED_SHORTCUT_OVERRIDE`), then restart the editor and the shortcut is no longer in the list.

**Comment by bruvzg** [comment]
> Now if a shortcut is loaded from the editor settings file (e.g. a user has customised a shortcut), then there is nothing to compare against for reverting it. No way to know what the 'original' was before customisation. This also breaks the shortcut list as it expects shortcuts to have an "original" meta set.

https://github.com/godotengine/godot/pull/73664 adds a new meta to mark user customized shortcuts (used only to skip overrides), and restores `set_meta("original"....` line.

**Pull request merged**